ci/test-sriov-secondary-network-aws.sh

```diff
@@ -563,10 +573,29 @@ function clean_up_all() {
       delete_subnet_cidr_reservation
 }
 
+function start_timeout_watcher() {
+    local timeout_seconds=$1
+    local parent_pid=$2
+
+    local safe_timeout=$((timeout_seconds - 300))
+
+    echo "Timeout watcher started. Will signal after ${safe_timeout} seconds."
+
+    sleep $safe_timeout
+
+    echo "Process timed out before AWS credential expiration! Sending termination signal to main process (PID: $parent_pid)"
+    kill -SIGTERM $parent_pid 2>/dev/null || true
+}
+
 echo "===========Test SR-IOV secondary network in AWS============="
 
+start_timeout_watcher "$AWS_DURATION_SECONDS" $$ &
+timeout_watcher_pid=$!
+
 if [[ "$RUN_SETUP_ONLY" != true ]]; then
-    trap clean_up_all EXIT
+    trap "clean_up_all && kill -9 $timeout_watcher_pid 2>/dev/null || true" EXIT
```

Comment:
If `clean_up_all` fails, `timeout_watcher_pid` won’t be killed, is that expected?